packages/google_maps_flutter/lib/src/marker_updates.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of google_maps_flutter;
+
+/// [Marker] update events to be applied to the [GoogleMap].
+///
+/// Used in [GoogleMapController] when the map is updated.
+class MarkerUpdates {
+  MarkerUpdates.internal({
+    @required this.markersToAdd,
+    @required this.markerIdsToRemove,
+    @required this.markersToChange,
+  })  : assert(markersToAdd != null),
+        assert(markerIdsToRemove != null),
+        assert(markersToChange != null);
+
+  /// Computes [MarkerUpdates] given previous and current [Marker]s.
+  factory MarkerUpdates.from(Set<Marker> previous, Set<Marker> current) {
```

Comment:
This always returns a new instance of MarkerUpdates, so no need to make it a factory constructor.